cnxdb/migrations/20170911201035_transform_cnxml_to_html.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: utf-8 -*-
+import logging
+import time
+from datetime import timedelta
+
+import rhaptos.cnxmlutils
+from dbmigrator import deferred, logger
+
+
+def _batcher(seq, size):
+    for pos in range(0, len(seq), size):
+         yield seq[pos:pos + size]
+
+
+def should_run(cursor, limit='LIMIT 1'):
+    version = rhaptos.cnxmlutils.__version__
+    version_text = '%data-cnxml-to-html-ver="{}"%'.format(version)
+    logger.info('Looking for {}'.format(version_text))
+    cursor.execute("""\
+WITH index_cnxml_html AS (
+    SELECT files.fileid, module_ident, file
+        FROM module_files NATURAL JOIN files
+        WHERE filename = 'index.cnxml.html'
+) SELECT fileid, max(module_ident)
+    FROM index_cnxml_html
+    WHERE convert_from(file, 'utf-8') NOT LIKE %s
+    GROUP BY fileid {}""".format(limit),
+                   (version_text,))
+    return cursor.fetchall()
+
+
+@deferred
+def up(cursor):
+    """Transform all index.cnxml to index.cnxml.html"""
+    # Get all the index.cnxml.html that does not have rhaptos.cnxmlutils
+    # current version
+    to_transform = should_run(cursor, limit='')
+    num_todo = len(to_transform)
+
+    batch_size = 10
```

Comment:
Running it now - batch size could be  a bit larger, if we wanted - I'm getting a line every 2-3 sec. Batch size of 100 would be acceptable (20-30 sec)